internal/config/profile_integration_test.go

```diff
@@ -0,0 +1,279 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build integration
+
+package config
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	atlasProfile   = "atlas"
+	newProfileName = "newProfileName"
+)
+
+func createProfile(profileContents string) error {
+	p.fs = afero.NewMemMapFs()
```

Comment:
I'm a bit worry that we are testing the global object and this could bring flakiness between tests, like if we change order or they interact with each other in some way we are not seeing (go tests also run in parallel)

I think I mentioned before that ideally each test should be testing its own `p` and this is doable
``` go
func TestProfile_Get_FullProfile(t *testing.T) {
  myLocalP := newProfile() // or even &profile{} if you want more control
  myLocalP.fs = afero.NewMemMapFs()
  myLocalP.Load()
  myLocalP.Get()
  // all my assertions
}
```

Reply:
I went back this way mostly to allow testing `Exists` and `List`, I can go either way on which one though

Reply:
We can ignore for now and see how it goes, if this starts to get flaky then we know what to do

Reply:
I was just testing unit & integration tests together and it started failing, probably something to do with this